modules/ROOT/pages/websocket/websocket-connector-config-topics.adoc

```diff
@@ -0,0 +1,285 @@
+</flow>
+----
+
+Notice the WebSocket server settings reference the HTTP listener configuration in the `listenerConfig` attribute. In this example, the WebSocket inbound listener listens for incoming messages on `localhost` and port 80, which are set in the HTTP listener configuration. This also sets basePath to `/ws` to override the HTTP basePath setting in the WebSocket configuration. To sum up, the WebSocket inbound listener listens for incoming messages at `+ws://localhost:80/ws/chat+`.
+
+Because the HTTP listener does not have a reference to a TLS configuration component, the connection opens by default using the WS protocol.
+
+== Configure Server Timeouts
+
+WebSocket connections are persistent and stateful. Therefore, it's a good practice to set a timeout to automatically close connections that have been idle.
+
+The following example shows how to use the `<websocket:server-settings>` to configure a 1-hour idle sockets timeout:
+
+[source,xml,linenums]
+----
+<http:listener-config name="httpListenerConfig" basePath="/regularHttp">
+   <http:listener-connection host="0.0.0.0" port="80" />
+</http:listener-config>
+
+<websocket:config name="wsServer">
+    <websocket:connection>
+       <websocket:server-settings
+         listenerConfig="httpListenerConfig" 
+         listenerBasePath="/ws" 
+         idleSocketTimeout="1" 
+         idleSocketTimeoutUnit="HOURS">
+    </websocket:connection>
+</websocket:config>
+----
+
+By default, all inbound sockets have an idle timeout of 15 minutes.
+
+== Configure WebSocket Client
+
+If your application needs to connect to external WebSocket services, provide the client settings element:
+
+[source,xml,linenums]
+----
+<websocket:config name="wsClient">
+    <websocket:connection>
+       <websocket:client-settings 
+         host="localhost" 
+         port="80" 
+         basePath="/ws" />
+    </websocket:connection>
+</websocket:config>
+
+<flow="connectToWebSocketFlow">
+    <websocket:open-outbound-socket 
+      path="/chat" 
+      socketId="myCustomWebSocketID-123" 
+      config-ref="wsClient" />
+    </websocket:open-outbound-socket>
+
+    <logger info="INFO" message="Opened connection to external service!"/>
+</flow>
+----
+
+In this example we have initialized an outbound socket which references the WebSocket client configuration ("wsClient") in the config-ref attribute. In this case, there is no need to reference an HTTP configuration.
+
+In the configuration above, we chose to specify a host, port, and basePath to build the WS request URI. The host parameter is optional, if not set, it is set to `localhost`. The port parameter is optional, if not specified, the default value is set according to the protocol scheme: the default for WS is 80, the default for WSS is 443. The basePath parameter is optional, if URI is not set, it indicates the base path for all client requests relative to the indicated host and port, if not set the default value will be set to `/`.
```

Comment:
```suggestion
In the configuration above, a `host`, `port`, and `basePath` are specified to build the WS request URI. The `host` parameter is optional; if not set it is set to "localhost". 

The `port` parameter is optional; if not specified the default value will be set according to the protocol scheme. If WS is used, the default value is set to `80`. If WSS is used, the default is set to `443`. 

The `basePath` parameter is optional; if the URI is not set, it will indicate the base path for all client requests relative to the indicated host and port. If the `basePath` parameter is not set, the default value is set to `/`.
```